File: bridges/nvda/addon/globalPlugins/nvdaMcpBridge/domain/ports/session_signals.py
# nvdaMcpBridge domain -- the SessionSignals port.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
# ROLE: domain port, audible cues that the bridge has taken or released NVDA, heard even in silent mode.
# USED BY: the Session controller.
# IMPLEMENTED BY: adapters/nvda_session_signals.py; tests/fakes/session_signals.py.

from __future__ import annotations

from abc import ABC, abstractmethod


class SessionSignals(ABC):
	@abstractmethod
	def session_started(self, persona: str) -> None:
		"""Two ascending tones, then ``persona`` spoken as received; empty means the tones alone."""

	@abstractmethod
	def session_ended(self) -> None:
		"""Two descending tones: control has been released."""

File: bridges/voiceover/Sources/VoiceOverBridgeDomain/Ports/SessionSignals.swift
// ROLE: port -- audible cues telling the human the bridge has taken or released control of their screen reader.
// IMPLEMENTED BY: AudibleSessionSignals, over the Tones seam and the Announcer port; ReportingSignals in the launcher; FakeSessionSignals.
// USED BY: the Session controller, once when a session establishes and once on the way out.
// A cue is a courtesy and never worth a session, so the Session guards every throwing cue.
// Cues play outside VoiceOver, because the suppression happens inside the capture voice.
public protocol SessionSignals: AnyObject {
	/// `persona` is announced as received; empty means the server declared none.
	func sessionStarted(persona: String) throws

	func sessionEnded() throws

	func silenceWarning() throws

	func silenceLifted() throws

	func silenceResuppressed() throws
}
